plugins/woocommerce/tests/php/src/Internal/AbilitiesApi/AbilitiesApiIntegrationTest.php

```
@@ -88,20 +79,30 @@ public function set_up() {
 		$this->original_init_action_count = $wp_actions['init'] ?? null;
 		$wp_actions['init']               = 1; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+
+		parent::setUp();
+
+		// Ensure Abilities API REST routes are registered if needed.
+		if ( function_exists( 'wp_register_ability' ) && class_exists( 'WP_REST_Abilities_Init' ) ) {
+			$routes = $this->server->get_routes();
+
+			// Check if any Abilities API routes are already registered.
+			$has_abilities_routes = false;
+			foreach ( $routes as $route => $handlers ) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$handlers'. (undefined)

(UnusedLocalVariable)